department-of-veterans-affairs/caseflow
# Adds CavcCorrespondenceMailTask

Resolves #15239 

### Description
Creates the CavcCorrespondenceMailTask & its behavior.

### Acceptance Criteria
- [ ] work behind the feature toggle: cavc_remand
- [ ] Add task should be accessible to the Mail Team
- [ ] accessible to the following user groups: Mail Team
- [ ] Select Correspondence Type drop down includes CAVC Correspondence
- [ ] requires a CAVC Remand appeal with an open CavcTask
- [ ] Mail Team users have no actions on the created tasks
- [ ] Cavc Lit Admins can act on Organization task or any User task
- [ ] Cavc Lit user can act on a user task assigned to themselves
- [ ]  Verify CavcRemandProcessedLetterResponseWindowTask status does not change.

### Testing Plan
  1. Reset the database with `make reset`
  2. Create a Cavc Appeal at CavcResponseWindow
```ruby
appeal = FactoryBot.create(:appeal, :cavc_response_window_open)
appeal.reload.treee
appeal.uuid
FeatureToggle.disable!(:cavc_remand)
```
  3. Load up [caseflow](http://localhost:3000/test/users) and sign in as Jolly Postman
  4. Navigate to the appeal you created
  5. Click `+ Add a new task`
   - [x] CAVC Correspondence is not available
  6. Enable the flag: `FeatureToggle.enable!(:cavc_remand)`
  7. Reload and Click `+ Add a new task`
   - [x] CAVC Correspondence is available
   - [x] Creating a CAVC Correspondence task autoassigns to CAVC Lit Support
   - [x] No actions are available on the task for the mail user
  8. Change to be a CAVC Lit Support admin user
  9. Navigate to the appeal
   - [x] Check the actions exist and do what you want (TK)
  10. Create another appeal & mail task
  11. Navigate to the appeal and assign it to a Cavc Lit support user
   - [x] confirm you have admin have actions on the user task
  12. Sign in as a Cavc Lit user you didn't assign the task to 
   - [x] No actions are available
  13. Sign in as the assigned Cavc Lit user
   - [x] Check the actions exist and do what you want (TK)

Expected Behavior on Actions:
  - [x] Confirm none of this affects the CavcResponseWindowTask status!
  - [x] Change Task Type: Cancels the CavcCorrespondence subtree, creates a sibling of the new type
  - [x] (Re)Assign to person: creates a user task
  - [x] Place on hold: places a hold
  - [x] Mark task complete: completes the subtree
  - [x] Cancel task: cancels the subtree
 
### Documentation Updates

### UI Changes
![image](https://user-images.githubusercontent.com/6129479/102548342-840b6000-4088-11eb-99ed-3aa1c1e6968b.png)
